### PR TITLE
fix: the tools.babel in rspack would cause a syntax error

### PR DESCRIPTION
**Pull request by GiveMe-A-Name**
## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-rspack-provider/tests/plugins/babel.test.ts [hunk 1]
```diff
@@ -2,14 +2,14 @@ import { describe, it, expect } from 'vitest';
 import { createBuilder } from '../helper';
 import { builderPluginBabel } from '@/plugins/babel';
 
-describe('plugins/babel', () => {
+describe.skip('plugins/babel', () => {
```
Comment (9aoy) [inline packages/builder/builder-rspack-provider/tests/plugins/babel.test.ts:5]: e2e 里面的 babel test 也要 skip 或者先 webpackOnly 吧

Comment (GiveMe-A-Name) [reply packages/builder/builder-rspack-provider/tests/plugins/babel.test.ts:5]: Got it

File: .changeset/weak-tables-compete.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@modern-js/builder-rspack-provider': patch
+---
+
+fix: the `tools.babel` in rspack would cause a syntax error
+fix: 在 rspack 中 `tools.babel` 会引起语法错误。
```
[the remark this answers is not in the document]
Comment (GiveMe-A-Name) [reply .changeset/weak-tables-compete.md]: Done~

File: packages/builder/builder-rspack-provider/tests/plugins/babel.test.ts (hunk 1 above)
Comment (GiveMe-A-Name) [reply packages/builder/builder-rspack-provider/tests/plugins/babel.test.ts:5]: Done~

**Review by 9aoy — APPROVED** [review]

**Pull request merged**